src/Services/LearnosityToQtiPreProcessingService.php

```diff
@@ -71,22 +73,26 @@ private function getFeatureReplacementString($node)
             $src = trim($node->attr['data-src']);
             $type = trim($node->attr['data-type']);
             if ($type === 'audioplayer' || $type === 'videoplayer') {
-                $src = $this->getSourceBasedOnMediaFormat($src);
-                return QtiMarshallerUtil::marshallValidQti(new ObjectElement($src, MimeUtil::guessMimeType(basename($src))));
+              $src = $this->getSourceBasedOnMediaFormat($src);
+              return QtiMarshallerUtil::marshallValidQti(new ObjectElement($src, MimeUtil::guessMimeType(basename($src))));
             }
         // Process regular question feature
         } else {
             $nodeClassAttribute = $node->attr['class'];
             $featureReference = $this->getFeatureReferenceFromClassName($nodeClassAttribute);
-            $feature = $this->widgets[$featureReference];
-            $type = $feature['data']['type'];
-
+            if (isset($this->widgets[$featureReference])) {
+                $feature = $this->widgets[$featureReference];
+                $type = isset($feature['data']['type']) ? $feature['data']['type'] : '';
+                $src = isset($feature['data']['src']) ? $feature['data']['src'] : '';
+            } else {
+                $feature = $this->widgets;
+                $type = isset($feature[1]['type']) ? $feature[1]['type'] : '';
+                $src = isset($feature[1]['src']) ? $feature[1]['src'] : '';
+            }
             if ($type === 'audioplayer' || $type === 'videoplayer') {
-                $src = $feature['data']['src'];
                 $object = new ObjectElement($src, MimeUtil::guessMimeType(basename($src)));
                 $object->setLabel($featureReference);
-                return QtiMarshallerUtil::marshallValidQti($object);
-
+                //return QtiMarshallerUtil::marshallValidQti($object);
```

Comment:
@pooja31chaudhary I'm confused by this - why are we no longer returning here? Is this not going to break conversion for `audioplayer` and `videoplayer`?

Reply:
Done

Reply:
@rakeshqdsa I can't tell what you've changed here. Are you sure you committed any changes to address this?

Reply:
Yes we have changed to handle widget feature in this file .

Reply:
@rakeshqdsa how exactly does this work? It seems to me like if the `type` is `audioplayer` or `videoplayer` and ends up in this block, the code will skip to the bottom and throw a `MappingException`. Am I missing something obvious here?

Reply:
Yes it will throw error.

Reply:
@rakeshqdsa Discussing this internally, it seems to me like we need to maintain support for `audioplayer` and `videoplayer`. This looks like it breaks support. Did you functionally test this against the content package that @michaelsharman provided?

Reply:
Yes this is supported .
I will generate a new PR for supporting these feature as well by EOD today .

Reply:
Hey @rakeshqdsa, it looks like the merge did something funny here - I can see two `else` branches for `sharedpassage`, one which is an `elseif` and one an `if` inside an `else`, which probably means it will stop at the first and throw an exception. Can you fix this up please?

https://github.com/Learnosity/learnosity-qti/pull/64/files#diff-381f8b249ceec79e819337439e64d46aR97
https://github.com/Learnosity/learnosity-qti/pull/64/files#diff-381f8b249ceec79e819337439e64d46aR103

Reply:
Done